src/BenchmarkManager.py

```diff
@@ -266,18 +266,66 @@ def load_config(self, config: dict, app_modules: list) -> None:
                         "devices"][single_device] = device_wrapper
 
     @staticmethod
-    def _query_for_config(config: dict, prefix: str = "") -> dict:
-        for key, config_answer in config.items():
+    def _query_for_config(param_opts: dict, prefix: str = "") -> dict:
+        config = {}
+        for key, config_answer in param_opts.items():
+            if not config_answer.get("if") is None:
+                #
+                #support parameter descriptions like
+                #"seed": {
+                #    "if": {"key":"graph_type", "in" : ["erdos-renyi"]},
+                #    ...
+                #}
+                #meaning that 'seed' only gets displayed if graph_type has been chosen to be 'erdos-renyi'
+                #This expects that the referenced parameter has been declared before and is declared to be
+                #'exclusive' so that its value is unique.
+                #
+
+                key_in_cond = config_answer.get("if")["key"]
+                dependency = param_opts.get(key_in_cond)
+
+                #check configuration is consistent
+                consistent = False
+                err_msg = None
+                if dependency is None:
+                    err_msg = f"Inconsistent parameter options: condition references unknown parameter: {key_in_cond}"
+                elif not dependency.get('exclusive', False):
+                    err_msg = f"Inconsistent parameter options: condition references non exclusive parameter: {key_in_cond}"
+                else:
+                    consistent = True
+                if not consistent:
+                    raise Exception(f"{prefix} {err_msg}")
+                    
+                if not config[key_in_cond][0] in config_answer.get("if")["in"]:
+                    continue
+                    
+            values = None
             if len(config_answer['values']) == 1:
                 # When there is only 1 value to choose from skip the user input for now
-                config[key] = config_answer['values']
+                values = config_answer['values']
+                print(f"{prefix} {config_answer['description']}: {config_answer['values'][0]}")
```

Comment:
Is this line needed? If yes` logging.info` or `logging.debug` should be used.

Reply:
I  think the print provides interesting information for the user. The fact that the parameter exists implies that it might have some other value and in fact if you edit the configuration file manually you can give it another value. So usually the user wants to know what the value is. Furthermore I think the information should be displayed in the same way as the other parameters are displayed - it is the same kind of information. So using print and not logging is appropriate in my opinion.  But  these arguments are not very strong so I'm fine with doing here whatever you prefer. Please just tell me what your preference is.

Reply:
I agree, I forgot that the inquirer is not using the logging to display its output.